File: app/src/main/java/com/lazylee/lzywanandroid/data/dao/TagDao.java
package com.lazylee.lzywanandroid.data.dao;

import com.lazylee.lzywanandroid.data.entity.Tag;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface TagDao {

    @Query("Select * From tag")
    List<Tag> getAll();

    List<Tag> getTagsByArticle(long articleId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Tag tag);
}

File: app/src/main/java/com/lazylee/lzywanandroid/data/dao/TagDao.kt
package com.lazylee.lzywanandroid.data.dao

import com.lazylee.lzywanandroid.data.entity.Tag

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface TagDao {

    @get:Query("Select * From tag")
    val all: List<Tag>

    fun getTagsByArticle(articleId: Long): List<Tag>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(tag: Tag)
}
